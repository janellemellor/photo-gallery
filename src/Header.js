import React, { Component } from "react";

export default class Header extends Component {
    render() {
        return (
            <header>
                <h2>Strange Creatures List</h2>
            </header>
        );
    }
}