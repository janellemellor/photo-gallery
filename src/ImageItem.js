import React, { Component } from "react";

export default class ImageItem extends Component {
    render() {

        return (
        <li>
           <div>
                <h3>{this.props.title}</h3>
                <img src={this.props.img} alt={this.props.description} />
            </div> 
        </li>
        
        );
    }
}