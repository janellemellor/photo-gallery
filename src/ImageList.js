import React, { Component } from "react";
import ImageItem from './ImageItem.js';


export default class ImageList extends Component {
    render() {
    return <ul>

    {this.props.images.map(image =>
        <ImageItem title={image.title} img={image.url} description={image.description} key={image.title} /> 
        )}

    </ul>
    }}
     