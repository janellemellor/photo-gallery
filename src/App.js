import React, { Component } from 'react';
import './App.css';
import images from './data.js';
import Header from './Header.js';
import ImageList from './ImageList.js';
import Select from './Select.js';


export default class App extends Component {
state = { selected: null };

  render() {
    const handleChange = e => {
      this.setState({ selected: e.target.value })};

    const filteredImages = images.filter(image => {
        if (!this.state.selected) return true;
        return image.keyword === this.state.selected; });
    
    return (
        <div>
          <Header />
          <Select handleChange={handleChange} />
          <ImageList images={filteredImages} />
         
        </div>
  )}}  