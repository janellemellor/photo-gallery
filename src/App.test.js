import React from 'react';
import images from './data.js'
import App from './App.js';
import Header from './Header.js';
import ImageList from './ImageList.js';
import ImageItem from './ImageItem.js';
import renderer from 'react-test-renderer';

test('renders App.js correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Header.js correctly', () => {
  const tree = renderer
  .create(<Header />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders ImageList.js correctly', () => {
  const tree = renderer
  .create(<ImageList images={images} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders ImageItem.js correctly', () => {
  const tree = renderer
  .create(<ImageItem />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});