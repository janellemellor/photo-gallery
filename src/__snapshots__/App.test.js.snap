// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App.js correctly 1`] = `
<div>
  <header>
    <h2>
      Strange Creatures List
    </h2>
  </header>
  <select
    onChange={[Function]}
  >
    <option
      defaultValue={true}
      value=""
    >
      All
    </option>
    <option
      value="narwhal"
    >
      Narwhal
    </option>
    <option
      value="rhino"
    >
      Rhino
    </option>
    <option
      value="unicorn"
    >
      Unicorn
    </option>
    <option
      value="unilego"
    >
      Unilego
    </option>
    <option
      value="triceratops"
    >
      Triceratops
    </option>
    <option
      value="markhor"
    >
      Markhor
    </option>
    <option
      value="mouflon"
    >
      Mouflon
    </option>
    <option
      value="addax"
    >
      Addax
    </option>
    <option
      value="chameleon"
    >
      Chameleon
    </option>
    <option
      value="lizard"
    >
      Lizard
    </option>
    <option
      value="dragon"
    >
      Dragon
    </option>
  </select>
  <ul>
    <li>
      <div>
        <h3>
          UniWhal
        </h3>
        <img
          alt="A unicorn and a narwhal nuzzling their horns"
          src="http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Rhino Family
        </h3>
        <img
          alt="Mother (or father) rhino with two babies"
          src="https://images.unsplash.com/photo-1512636618879-bbe79107e9e3?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=bd9460ee6d1ddbb6b1ca7be86dfc4590&auto=format&fit=crop&w=1825&q=80"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Unicorn Head
        </h3>
        <img
          alt="Someone wearing a creepy unicorn head mask"
          src="https://www.dhresource.com/0x0s/f2-albu-g5-M00-1A-11-rBVaI1hsIIiALxKzAAIHjSU3VkE490.jpg/wholesale-halloween-costume-prop-unicorn.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          UniLego
        </h3>
        <img
          alt="Lego figurine dressed in a unicorn outfit"
          src="https://images.unsplash.com/photo-1518946222227-364f22132616?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=4836a6fca62e7dce9324346bacfde085&auto=format&fit=crop&w=2534&q=80"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Basically a unicorn
        </h3>
        <img
          alt="A narwhal is basically a unicorn after all, right?"
          src="https://i.pinimg.com/736x/b4/61/06/b46106830b841017ea59870b27ec18dc--narwhals-a-unicorn.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          #truth
        </h3>
        <img
          alt="The truth behind narwhals"
          src="https://i.pinimg.com/originals/16/cf/2a/16cf2a0b3fd51b9bee08bb6296193b75.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Baby Rhino
        </h3>
        <img
          alt="This is actually a figurine but it looks kinda real"
          src="https://secure.img1-ag.wfcdn.com/im/17007094/resize-h800%5Ecompr-r85/3589/35892451/Baby+Rhino+Figurine.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Cera
        </h3>
        <img
          alt="Three horns but still, horns. And who doesn't like The Land Before Time?"
          src="https://vignette.wikia.nocookie.net/landbeforetime/images/c/c3/Cera_infobox.png/revision/latest?cb=20180422005003"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Narwhal costume
        </h3>
        <img
          alt="A woman wearing a blue narwhal costume"
          src="https://ae01.alicdn.com/kf/HTB18GwSQVXXXXaZaXXXq6xXFXXXh/Animal-Cosplay-Costume-Narwhal-Onesie-Mens-Womens-Cartoon-Whale-Pajamas.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Rhino costume
        </h3>
        <img
          alt="Mascots have to get their costumes somewhere"
          src="https://www.shopmascot.com/image/cache/mascotnew/new196-800x800.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Believe
        </h3>
        <img
          alt="I believe in unicorns, do you?"
          src="https://www.tinselbox.com/wp-content/uploads/2018/03/I-BELIEVE-IN-UNICORNS-FREE-PRINTABLE-WATERCOLOR-7-of-1.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Markhor
        </h3>
        <img
          alt="These wild goats eat snakes, then secrete a foam that locals fight over for the antivenom properties -- how cool is that?"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/38/Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg/220px-Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Baby markhor
        </h3>
        <img
          alt="Even the babies are adorable"
          src="http://www.zooborns.com/.a/6a010535647bf3970b0223c84d5959200c-800wi"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Mouflon
        </h3>
        <img
          alt="Those horns though"
          src="http://cdn.modernfarmer.com/wp-content/uploads/2014/08/28476658_9c97f35096_o.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Addax
        </h3>
        <img
          alt="This guy is basically extinct but survives well in captivity, so they're frequently found in zoos"
          src="http://cdn.modernfarmer.com/wp-content/uploads/2014/08/addax.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Baby mouflon
        </h3>
        <img
          alt="The cuteness that is a baby mouflon asleep"
          src="https://cbsnews3.cbsistatic.com/hub/i/r/2013/03/05/5b414225-a645-11e2-a3f0-029118418759/thumbnail/620x350/2d4cf24685b45c22912e64d2004fec8d/Baby_Mouflon_Wild_Sheep.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Happy Jackson's Chameleon
        </h3>
        <img
          alt="These are really common in Hawaii"
          src="http://cdn.modernfarmer.com/wp-content/uploads/2014/08/chameleon.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Serious Jackson's Chameleon
        </h3>
        <img
          alt="This one is very serious."
          src="https://imgc.allpostersimages.com/img/print/posters/dlillc-jackson-s-chameleon_a-G-13448768-14258384.jpg"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Horned Lizard
        </h3>
        <img
          alt="Fave food: ants"
          src="https://www.nps.gov/band/learn/nature/images/short-horned-lizard-open-mouth-18.jpg?maxwidth=650&autorotate=false"
        />
      </div>
    </li>
    <li>
      <div>
        <h3>
          Smaug
        </h3>
        <img
          alt="Fan illustration of Smaug from 'The Hobbit'"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/df/Smaug_par_David_Demaret.jpg/290px-Smaug_par_David_Demaret.jpg"
        />
      </div>
    </li>
  </ul>
</div>
`;

exports[`renders Header.js correctly 1`] = `
<header>
  <h2>
    Strange Creatures List
  </h2>
</header>
`;

exports[`renders ImageItem.js correctly 1`] = `
<li>
  <div>
    <h3 />
    <img />
  </div>
</li>
`;

exports[`renders ImageList.js correctly 1`] = `
<ul>
  <li>
    <div>
      <h3>
        UniWhal
      </h3>
      <img
        alt="A unicorn and a narwhal nuzzling their horns"
        src="http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Rhino Family
      </h3>
      <img
        alt="Mother (or father) rhino with two babies"
        src="https://images.unsplash.com/photo-1512636618879-bbe79107e9e3?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=bd9460ee6d1ddbb6b1ca7be86dfc4590&auto=format&fit=crop&w=1825&q=80"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Unicorn Head
      </h3>
      <img
        alt="Someone wearing a creepy unicorn head mask"
        src="https://www.dhresource.com/0x0s/f2-albu-g5-M00-1A-11-rBVaI1hsIIiALxKzAAIHjSU3VkE490.jpg/wholesale-halloween-costume-prop-unicorn.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        UniLego
      </h3>
      <img
        alt="Lego figurine dressed in a unicorn outfit"
        src="https://images.unsplash.com/photo-1518946222227-364f22132616?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=4836a6fca62e7dce9324346bacfde085&auto=format&fit=crop&w=2534&q=80"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Basically a unicorn
      </h3>
      <img
        alt="A narwhal is basically a unicorn after all, right?"
        src="https://i.pinimg.com/736x/b4/61/06/b46106830b841017ea59870b27ec18dc--narwhals-a-unicorn.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        #truth
      </h3>
      <img
        alt="The truth behind narwhals"
        src="https://i.pinimg.com/originals/16/cf/2a/16cf2a0b3fd51b9bee08bb6296193b75.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Baby Rhino
      </h3>
      <img
        alt="This is actually a figurine but it looks kinda real"
        src="https://secure.img1-ag.wfcdn.com/im/17007094/resize-h800%5Ecompr-r85/3589/35892451/Baby+Rhino+Figurine.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Cera
      </h3>
      <img
        alt="Three horns but still, horns. And who doesn't like The Land Before Time?"
        src="https://vignette.wikia.nocookie.net/landbeforetime/images/c/c3/Cera_infobox.png/revision/latest?cb=20180422005003"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Narwhal costume
      </h3>
      <img
        alt="A woman wearing a blue narwhal costume"
        src="https://ae01.alicdn.com/kf/HTB18GwSQVXXXXaZaXXXq6xXFXXXh/Animal-Cosplay-Costume-Narwhal-Onesie-Mens-Womens-Cartoon-Whale-Pajamas.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Rhino costume
      </h3>
      <img
        alt="Mascots have to get their costumes somewhere"
        src="https://www.shopmascot.com/image/cache/mascotnew/new196-800x800.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Believe
      </h3>
      <img
        alt="I believe in unicorns, do you?"
        src="https://www.tinselbox.com/wp-content/uploads/2018/03/I-BELIEVE-IN-UNICORNS-FREE-PRINTABLE-WATERCOLOR-7-of-1.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Markhor
      </h3>
      <img
        alt="These wild goats eat snakes, then secrete a foam that locals fight over for the antivenom properties -- how cool is that?"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/38/Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg/220px-Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Baby markhor
      </h3>
      <img
        alt="Even the babies are adorable"
        src="http://www.zooborns.com/.a/6a010535647bf3970b0223c84d5959200c-800wi"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Mouflon
      </h3>
      <img
        alt="Those horns though"
        src="http://cdn.modernfarmer.com/wp-content/uploads/2014/08/28476658_9c97f35096_o.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Addax
      </h3>
      <img
        alt="This guy is basically extinct but survives well in captivity, so they're frequently found in zoos"
        src="http://cdn.modernfarmer.com/wp-content/uploads/2014/08/addax.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Baby mouflon
      </h3>
      <img
        alt="The cuteness that is a baby mouflon asleep"
        src="https://cbsnews3.cbsistatic.com/hub/i/r/2013/03/05/5b414225-a645-11e2-a3f0-029118418759/thumbnail/620x350/2d4cf24685b45c22912e64d2004fec8d/Baby_Mouflon_Wild_Sheep.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Happy Jackson's Chameleon
      </h3>
      <img
        alt="These are really common in Hawaii"
        src="http://cdn.modernfarmer.com/wp-content/uploads/2014/08/chameleon.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Serious Jackson's Chameleon
      </h3>
      <img
        alt="This one is very serious."
        src="https://imgc.allpostersimages.com/img/print/posters/dlillc-jackson-s-chameleon_a-G-13448768-14258384.jpg"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Horned Lizard
      </h3>
      <img
        alt="Fave food: ants"
        src="https://www.nps.gov/band/learn/nature/images/short-horned-lizard-open-mouth-18.jpg?maxwidth=650&autorotate=false"
      />
    </div>
  </li>
  <li>
    <div>
      <h3>
        Smaug
      </h3>
      <img
        alt="Fan illustration of Smaug from 'The Hobbit'"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/df/Smaug_par_David_Demaret.jpg/290px-Smaug_par_David_Demaret.jpg"
      />
    </div>
  </li>
</ul>
`;

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
